File: system-fe/lib/api/spaceStation.ts
import apiClient from './client';

export interface SpaceStation {
  id: number;
  name: string;
  systemId: number;
  systemName: string;
  playerId: string;
  playerName: string;
  ironAmount: number;
  copperAmount: number;
  fuelAmount: number;
  soilAmount: number;
  createdAt: string;
}

export async function getPlayerSpaceStations(gameId: number): Promise<SpaceStation[]> {
  const response = await apiClient.get<SpaceStation[]>(`/api/space-stations/game/${gameId}`);
  return response.data;
}

export async function getSystemSpaceStations(systemId: number): Promise<SpaceStation[]> {
  const response = await apiClient.get<SpaceStation[]>(`/api/space-stations/system/${systemId}`);
  return response.data;
}

export async function getSpaceStation(id: number): Promise<SpaceStation> {
  const response = await apiClient.get<SpaceStation>(`/api/space-stations/${id}`);
  return response.data;
}

File: system-be/SystemGame.Api/Data/Entities/SpaceStation.cs
namespace SystemGame.Api.Data.Entities;

public class SpaceStation
{
    public int Id { get; set; }
    public int SystemId { get; set; }
    public string PlayerId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Resource storage
    public double IronAmount { get; set; } = 0;
    public double CopperAmount { get; set; } = 0;
    public double FuelAmount { get; set; } = 0;
    public double SoilAmount { get; set; } = 0;
    
    // Navigation properties
    public virtual StarSystem StarSystem { get; set; } = null!;
    public virtual AppUser Player { get; set; } = null!;
}

